src/Diagnostics/FxCop/System.Runtime.Analyzers/VisualBasic/Globalization/UseOrdinalStringComparison.Fixer.vb

```diff
@@ -0,0 +1,73 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System.Composition
+Imports System.Threading
+Imports System.Threading.Tasks
+Imports Microsoft.CodeAnalysis
+Imports Microsoft.CodeAnalysis.CodeFixes
+Imports Microsoft.CodeAnalysis.Editing
+Imports Microsoft.CodeAnalysis.Formatting
+Imports Microsoft.CodeAnalysis.Shared.Extensions
+Imports Microsoft.CodeAnalysis.VisualBasic
+Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
+
+Namespace System.Runtime.Analyzers
+    <ExportCodeFixProvider(LanguageNames.VisualBasic), [Shared]>
+    Public Class BasicUseOrdinalStringComparisonFixer
+        Inherits UseOrdinalStringComparisonFixerBase
+
+        Protected Overrides Function IsInArgumentContext(node As SyntaxNode) As Boolean
+            Return node.IsKind(SyntaxKind.SimpleArgument) AndAlso
+                   Not DirectCast(node, SimpleArgumentSyntax).IsNamed AndAlso
+                   DirectCast(node, SimpleArgumentSyntax).Expression.IsKind(SyntaxKind.SimpleMemberAccessExpression)
+        End Function
+
+        Protected Overrides Function FixArgument(document As Document, syntaxFactoryService As SyntaxGenerator, root As SyntaxNode, argument As SyntaxNode) As Task(Of Document)
+            Dim memberAccess = TryCast(TryCast(argument, SimpleArgumentSyntax)?.Expression, MemberAccessExpressionSyntax)
+            If memberAccess IsNot Nothing Then
+                ' preserve the "IgnoreCase" suffix if present
+                Dim isIgnoreCase = memberAccess.Name.GetText().ToString().EndsWith(UseOrdinalStringComparisonAnalyzer.IgnoreCaseText, StringComparison.Ordinal)
+                Dim newOrdinalText = If(isIgnoreCase, UseOrdinalStringComparisonAnalyzer.OrdinalIgnoreCaseText, UseOrdinalStringComparisonAnalyzer.OrdinalText)
+                Dim newIdentifier = syntaxFactoryService.IdentifierName(newOrdinalText)
+                Dim newMemberAccess = memberAccess.WithName(CType(newIdentifier, SimpleNameSyntax)).WithAdditionalAnnotations(Formatter.Annotation)
+                Dim newRoot = root.ReplaceNode(memberAccess, newMemberAccess)
+                Return Task.FromResult(document.WithSyntaxRoot(newRoot))
+            End If
+            Return Task.FromResult(document)
```

Comment:
then, shouldn't it actually be async then. Task.FromResult means, it does all work upfront and pretend it is lazy.